specs/refactoring_of_node_monitoring_flows_into_node_agent.adoc

```diff
@@ -0,0 +1,232 @@
+// vim: tw=79
+
+= Refactoring Of Node Monitoring Flows Into Node-Agent
+
+== Introduction
+
+This specification deals with detailing the work involved in merging the
+node-monitoring and node-agent services.
+
+Currently the responsibilities of node-monitoring are as follows:
+
+* It packages the collectd custom plugins
+    ** sds-specific utilization, status and counters plugins
+    ** node cluster network throughput plugin
+* Jinja2 templates for configuring the collectd plugins(both custom and
+  collectd provided)
+* A generic flow and a python script(that can be used as a linux command) to
+  get the configs generated from jinja templates given the appropriate parameters.
+
+== Problem description
+
+The idea behind having node-agent and node-monitoring seperated into a seperate
+packages was to facilitate modularity in tendrl stack such that users of tendrl
+stack can have the freedom to choose whether or not to have monitoring stack
+along with tendrl core stack. But, this pulls in overhead of having to maintain
+greater number of services.So, the idea is to now consolidate the services as
+much as possible and as part of this idea, this spec proposes consolidation of
+node-monitoring into node-agent.
+
+== Use Cases
+
+Admin creates/imports a cluster using/into tendrl and has monitoring enabled
+out of the box.
+
+== Proposed change
+
+* The generic flow as in:
+
+----
+https://github.com/Tendrl/node-monitoring/blob/master/tendrl/node_monitoring/flows/configure_collectd/__init__.py
+----
+
+and a genric command to generate config file from template as in:
+
+----
+https://github.com/Tendrl/node-monitoring/blob/master/tendrl/node_monitoring/commands/config_manager.py
+----
+
+will be moved to node-agent.
+
+* The templates currently in:
+
+----
+https://github.com/Tendrl/node-monitoring/tree/master/tendrl/node_monitoring/templates
+----
+
+and the plugins currently in:
+
+----
+https://github.com/Tendrl/node-monitoring/tree/master/tendrl/node_monitoring/plugins
+----
+
+will be moved into node-agent.
+
+* Earlier the jobs for intiating the flows to configure collectd on nodes were
+  initiated by performance-monitoring node. But for now(momentarily) as a final
+  step of import, create and expand cluster, this will be intiated on all nodes
+  of cluster.
+
+* Initiating(loading job to etcd) configuration on cluster nodes will involve
+  invoking configuration flow from an atom marked post atom in import cluster
+  flow.
+  ----
+  Job loading sample:
+  https://github.com/Tendrl/performance-monitoring/blob/master/tendrl/performance_monitoring/utils/util.py#L41
+  ----
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+A new flow definition will be added to node-agent under its namespace as below:
+
+----
+namespace.node_agent:
+  flows:
+    ConfigureCollectd:
+      atoms:
+        - tendrl.objects.Node.atoms.Cmd
```

Comment:
any reason why this could be incorrect because I have only copy pasted the definition entry as it used to be nnode-monitoring def file(which used to work during our previous drops) and only refactored node-monitoring=>node-agent